### PR TITLE
Bunny ears now adapt to the hair color of the wearer

### PR DESCRIPTION
**Pull request by ivanmixo**
## About The Pull Request

Title. Moved the behaviour to a component. 

## Why It's Good For The Game

Why not have them adapt to the wearer's hair color?

## Changelog
:cl:
add: Rabbit ears now adapt to the wearer's hairstyle like kitty ears do
/:cl:

### CODE REVIEW

**Comment by francinum** [comment]
~~It'd probably be best for you to make them both a subtype of an abstract 'hairmatch' clothing object, or move the behavior to a component.~~

**Comment by ivanmixo** [comment]
I could do a component I guess, might as well.

**Comment by Effemu** [comment]
obraz

You will be punished for your sins.

**Comment by ivanmixo** [comment]
> ~It'd probably be best for you to make them both a subtype of an abstract 'hairmatch' clothing object, or move the behavior to a component.~

Done

**Comment by Wilchenx** [comment]
gateway to adding viera

**Comment by ivanmixo** [comment]
> 
> 
> 
> ## About The Pull Request
> 
> Title. Moved the kitty and bunny ear functionality to a component.
> 
> 
> 
> ## Why It's Good For The Game
> 
> Why not have them adapt to the wearer's hair color?
> 
> 
> 
> ## Changelog
> :cl:
> add: Rabbit ears now adapt to the wearer's hairstyle like kitty ears do
> /:cl:
> 
> 
> 
>

**Review by francinum — DISMISSED** [review]

**Comment by ToasterBan** [comment]
> gateway to adding viera

stop using weeb words and say bunnygirls like a normal person

File: code/datums/components/haircolor_clothes.dm [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/datum/component/haircolor_clothing/Initialize()
+    RegisterSignal(parent, COMSIG_ITEM_EQUIPPED, .proc/onEquip)
+    RegisterSignal(parent, COMSIG_ATOM_UPDATE_ICON, .proc/update_color)
```
Comment (Archanial) [inline code/datums/components/haircolor_clothes.dm:3]: unregister on remove

**Review by Archanial — DISMISSED** [review]

File: code/datums/components/haircolor_clothes.dm (hunk 1 above)
Comment (ivanmixo) [reply code/datums/components/haircolor_clothes.dm:3]: Look at `datum/Destroy()` please

**Comment by francinum** [comment]
I'm not sure if the destruction thing needs to be touched.

**Comment by ivanmixo** [comment]
99% sure `datum/Destroy()` clears all the signals and components' signals anyways. Don't think I need to manually unregister them.

Was wrong haha.

**Comment by francinum** [comment]
Signal refs and all attached components are automatically garbage collected as part of datum level destroy. Handling it yourself is nice for cleanliness.

**Review by francinum — APPROVED** [review]

**Pull request merged**